### PR TITLE
CORE-395: Linux compilation issues

### PR DESCRIPTION
**Pull request by EBGToo**

### CODE REVIEW

File: WalletKitCore/crypto/BRCryptoAccount.c [hunk 1]
```diff
@@ -57,7 +54,7 @@ cryptoAccountDeriveSeed (const char *phrase) {
 extern char *
 cryptoAccountGeneratePaperKey (const char *words[]) {
     UInt128 entropy;
-    randomBytes (entropy.u8, sizeof(entropy));
+    random_bytes_brd (entropy.u8, sizeof(entropy));
```
Comment (michael-brd) [inline WalletKitCore/crypto/BRCryptoAccount.c]: If we are going with the suffix, I would think we'd have this be `arc4random_buf_brd`?

Comment (EBGToo) [reply WalletKitCore/crypto/BRCryptoAccount.c]: Fair enough.

File: WalletKitCore/support/BROSCompat.c [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+#elif defined (__APPLE__) // IOS, MacOS
+    if (0 != SecRandomCopyBytes(kSecRandomDefault, bytesCount, bytes))
+        arc4random_buf (bytes, bytesCount); // fallback
+#else
+#  error Undefined random_bytes_brd()
+#endif
+}
+
+#if defined (__linux__) && !defined(strlcpy)
```
Comment (michael-brd) [inline WalletKitCore/support/BROSCompat.c]: this is interesting; I didn't get issues here when I built `corenative-jre` for linux. I wonder if we were just missing the header? Or you were getting an unresolved symbol on link?

File: Package.swift [hunk 3]
```diff
@@ -44,7 +44,9 @@ let package = Package(
                 .headerSearchPath("vendor/secp256k1"),
             ],
             linkerSettings: [
-                .linkedLibrary("resolv")
+                .linkedLibrary("resolv"),
+//                .linkedLibrary("bsd"),
```
Comment (michael-brd) [inline Package.swift]: we need libbsd, no? I needed it on linux due to `nftw`

File: WalletKitCore/crypto/BRCryptoWalletManager.c [hunk 4]
```diff
@@ -31,6 +31,8 @@
 #include "ethereum/BREthereum.h"
 #include "support/BRFileService.h"
 
+uint64_t BLOCK_HEIGHT_UNBOUND_VALUE = UINT64_MAX;
```
Comment (michael-brd) [inline WalletKitCore/crypto/BRCryptoWalletManager.c]: Also interesting, didn't have an issue with this one personally for `corenative-jre` on linux.

File: Package.swift (hunk 3 above)
Comment (EBGToo) [reply Package.swift]: It is required on linux; but I can't get Package.swift to conditionally include it.  I just don't know the syntax.

File: WalletKitCore/crypto/BRCryptoWalletManager.c (hunk 4 above)
Comment (EBGToo) [reply WalletKitCore/crypto/BRCryptoWalletManager.c]: The linux `swift build` refused to see `#define BLOCK_HEIGHT...`

File: WalletKitCore/support/BROSCompat.h [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+extern void
+arc4random_buf_brd (void *bytes, size_t bytesCount);
+
+extern uint32_t
+arc4random_uniform_brd(uint32_t upperBbound);
+
+#if defined (__linux__) && !defined(strlcpy)
+extern size_t
+strlcpy (char *dst, const char *src, size_t siz);
+#endif
```
Comment (michael-brd) [inline WalletKitCore/support/BROSCompat.h]: Is this needed?

Comment (EBGToo) [reply WalletKitCore/support/BROSCompat.h]: Sometimes 'hey idiot' is more appropriate! :-)

File: WalletKitCoreTests/test/support/testSup.c [hunk 6]
```diff
@@ -7,35 +7,18 @@
 //
 
 #include <stdio.h>
-#include <pthread.h>
 #include <sys/stat.h>
 #include <sys/time.h>
+#define __USE_XOPEN_EXTENDED
```
Comment (michael-brd) [inline WalletKitCoreTests/test/support/testSup.c]: [no change required] I have no issue with this to move things forward but; is this the right way to do this? Geniune question. Never knew the answer for these compatibility macros

Comment (EBGToo) [reply WalletKitCoreTests/test/support/testSup.c]: Nor I so just moving it forward.  Maybe using .unsafeFlags() in Package.swift could handle it.

**Pull request merged**